pkg/autoupgrade/daemon_test.go

```diff
@@ -337,6 +341,30 @@ func TestRefreshImages(t *testing.T) {
 			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "notify-app"): now},
 			appsUpdated:            map[string]string{"notify-app": "docker.io/acorn/notify"},
 		},
+		{
+			name:                   "Auto refresh tag with multiple remote tags with latest tag image denied and current being an older image",
+			client:                 &mockDaemonClient{remoteTags: []string{"v1.1.1", "v1.1.2"}, imageDenyList: map[string]struct{}{"index.docker.io/acorn/test-1:v1.1.2": {}}},
+			appKeysPrevCheckBefore: map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): thirtySecondsAgo},
+			imagesToRefresh:        map[imageAndNamespaceKey][]kclient.ObjectKey{{image: "index.docker.io/acorn/test-1:v1.1.1", namespace: "acorn"}: {router.Key("acorn", "test-1")}},
+			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): now},
+			appsUpdated:            map[string]string{"test-1": "index.docker.io/acorn/test-1:v1.1.1"},
```

Comment:
This is saying that you expect the app `test-1` to be updated to `v1.1.1`, but it is already at `v1.1.1`. So, we want **no** update to the app, right? This `appsUpdated` map should be empty in this test case.